### PR TITLE
Allow archiving projects via Project Metadata

### PR DESCRIPTION
**Pull request by peripheralist**
## What does this PR do and why?

Addresses #587 

Allows project owners to archive/unarchive their projects via the project metadata, which also automatically notifies us via Github to update `archivedProjectIds`.

- New ProjectMetadataV4 with `archived` property. Projects will appear archived if their `metadata.archived === true` OR if `archivedProjectIds.includes(<id>)`
- New archive/unarchive functionality in project tools drawer. This will update the project's metadata, and automatically create a Github issue to notify us to update the `archivedProjectIds` as needed (see #599 & #600)
- Adds Octokit integration to allow automating creation of github issues. _Note:_ This requires a new .env var `REACT_APP_GITHUB_ACCESS_TOKEN`

## Screenshots or screen recordings

image
image

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](../CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](../CONTRIBUTING.md#browser-support).

### CODE REVIEW

File: src/constants/octokit.ts
Comment (peripheralist) [inline src/constants/octokit.ts:4]: We may want to throw an error on load if the .env var isn't set

**Review by johnnyd-eth — APPROVED** [review]
This is so sick @peripheralist, absolutely key update. This flow is clearly a massive improvement from what we've got, and I can't think of how we could make it any easier for ourselves and project owners than this. I left one minor comment but I'm approving in advance ✅

File: src/components/v1/V1Project/modals/ProjectToolDrawerModal.tsx
Comment (peripheralist) [reply src/components/v1/V1Project/modals/ProjectToolDrawerModal.tsx:304]: ohh good catch, wrote this off the cuff. should probably be "Juicebox protocol" actually

File: src/components/v1/V1Project/modals/ProjectToolDrawerModal.tsx
Comment (aeolianeth) [inline src/components/v1/V1Project/modals/ProjectToolDrawerModal.tsx:113]: **suggestion**: I imagine this might take markdown, wondering if this works:

```suggestion
            body: `**Chain:** ${readNetwork.name}
            **Project handle:** ${handle}
            **Project ID:** ${projectId?.toString()}`,
```

File: src/components/v1/V1Project/modals/ProjectToolDrawerModal.tsx
Comment (aeolianeth) [inline src/components/v1/V1Project/modals/ProjectToolDrawerModal.tsx:107]: **suggestion (non-blocking)** would be cool to add a github label although couldn't find docs for how to do that

File: src/components/v1/V1Project/modals/ProjectToolDrawerModal.tsx
Comment (aeolianeth) [inline src/components/v1/V1Project/modals/ProjectToolDrawerModal.tsx:304]: **suggestion** can we wrap these strings in `Trans` tags?

**Review by aeolianeth — CHANGES_REQUESTED** [review]
@peripheralist love it! The github issue workflow feels very good, I think this will scale well into the medium term.

I left a few minor comments, but I'm most interested in your thoughts on potentially tackling this a different way to avoid adding a dependency. LMK! 🙏 

cc @johnnyd-eth

File: src/constants/octokit.ts
Comment (peripheralist) [reply src/constants/octokit.ts:3]: agreed. feels meh in constants, but couldn't find a better place to put it. to me this feels logically similar to ReadProvider which is in the same dir, so 🤷

File: src/constants/octokit.ts
Comment (peripheralist) [reply src/constants/octokit.ts:1]: good looking out here. looks like the package is only 52kB so pretty tiny https://www.npmjs.com/package/@octokit/core. kinda odd, we could only depend on this package https://www.npmjs.com/package/@octokit/request which seems like it's supposed to be slimmer, but its size is actualy larger at 57kB... 🤔 

i'm not sure the best way to make that tree shaking work. currently the ProjectToolDrawer itself is loaded even on the app landing page lol... 
image

File: src/components/v1/V1Project/modals/ProjectToolDrawerModal.tsx
Comment (peripheralist) [reply src/components/v1/V1Project/modals/ProjectToolDrawerModal.tsx:304]: thank you!!

File: src/components/v1/V1Project/modals/ProjectToolDrawerModal.tsx
Comment (peripheralist) [reply src/components/v1/V1Project/modals/ProjectToolDrawerModal.tsx:113]: hm... really thought that would work

image

image

File: src/constants/octokit.ts
Comment (aeolianeth) [reply src/constants/octokit.ts:1]: Sweet thanks for looking into it. I think I would still much prefer this lives in a script outside of the app (github action triggered by a POST request feels good to me), but also want to keep a bias towards getting this available to users. Happy to leave this up to you 🙏

File: src/components/v1/V1Project/modals/ProjectToolDrawerModal.tsx
Comment (aeolianeth) [reply src/components/v1/V1Project/modals/ProjectToolDrawerModal.tsx:113]: LAME. All good haha, thanks for indulging me 😊

File: src/components/v1/V1Project/modals/ProjectToolDrawerModal.tsx
Comment (aeolianeth) [reply src/components/v1/V1Project/modals/ProjectToolDrawerModal.tsx:107]: yeah love either. We can slap whatever we use on the Crowdin PRs too

File: src/components/v1/V1Project/modals/ProjectToolDrawerModal.tsx
Comment (aeolianeth) [reply src/components/v1/V1Project/modals/ProjectToolDrawerModal.tsx]: I think there's a way we can lint for missing translations 👀 . Will look into it e v e n t u a l l y

File: src/constants/octokit.ts
Comment (peripheralist) [reply src/constants/octokit.ts:1]: > Sweet thanks for looking into it. I think I would still much prefer this lives in a script outside of the app (github action triggered by a POST request feels good to me), but also want to keep a bias towards getting this available to users. Happy to leave this up to you 🙏

great suggestion! agreed, let me try to get that working.

**Comment by peripheralist** [comment]
update: was able to remove the Octokit dependency and just use a simple `axios.post()` to create github issue. (i tried this first, but must have messed something up the first time)

other important change—i've moved the issue creation step _before_ the `setUri` tx is created. this ensures the issue gets created even if a user leaves the page while the tx is pending. the tradeoff is that it's now possible for an issue to be created for a project whose archive tx didn't complete, so we'll just wanna double check all projects that have archive/unarchive issues created to make sure that their metadata was actually updated.

File: src/utils/ipfs.ts
Comment (peripheralist) [inline src/utils/ipfs.ts:276]: passing improvement, just ensures that anytime metadata is uploaded it's converted to the latest version

**Pull request merged**
